src/checkout-buttons/strategies/googlepay/googlepay-authorizenet-button-strategy.spec.ts

```diff
@@ -0,0 +1,180 @@
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+            .mockReturnValue(paymentMethod);
+        jest.spyOn(paymentProcessor, 'initialize')
+            .mockReturnValue(Promise.resolve());
+        jest.spyOn(paymentProcessor, 'deinitialize');
+
+        walletButton = document.createElement('a');
+        walletButton.setAttribute('id', 'mockButton');
+        jest.spyOn(paymentProcessor, 'createButton')
+            .mockReturnValue(walletButton);
+
+        container = document.createElement('div');
+        container.setAttribute('id', 'googlePayCheckoutButton');
+        document.body.appendChild(container);
+    });
+
+    afterEach(() => {
+        document.body.removeChild(container);
+    });
+
+    it('creates an instance of GooglePayButtonStrategy', () => {
+        expect(strategy).toBeInstanceOf(GooglePayButtonStrategy);
+    });
+
+    describe('#initialize()', () => {
+        describe('Payment method exist', () => {
+            it('Creates the button', async () => {
+                checkoutButtonOptions = getCheckoutButtonOptions();
+
+                await strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.createButton).toHaveBeenCalled();
+            });
+
+            it('Validates if strategy has been initialized', async () => {
+                checkoutButtonOptions = getCheckoutButtonOptions();
+
+                await strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.initialize).toHaveBeenCalledTimes(1);
+            });
+
+            it('fails to initialize the strategy if no container id is supplied', () => {
+                checkoutButtonOptions = getCheckoutButtonOptions(Mode.UndefinedContainer);
+
+                function initialize() {
+                    strategy.initialize(checkoutButtonOptions);
+                }
+
+                expect(initialize).toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if no valid container id is supplied', () => {
+                checkoutButtonOptions = getCheckoutButtonOptions(Mode.InvalidContainer);
+
+                return expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(InvalidArgumentError);
+            });
+        });
+    });
+
+    describe('#deinitialize()', () => {
+        let containerId: string;
+
+        beforeAll(() => {
+            checkoutButtonOptions = getCheckoutButtonOptions(Mode.GooglePayAuthorizeNet);
+            containerId = checkoutButtonOptions.containerId;
+        });
+
+        it('check if googlepay payment processor deinitialize is called', async () => {
+            await strategy.initialize(checkoutButtonOptions);
+
+            await strategy.deinitialize();
+
+            expect(paymentProcessor.deinitialize).toBeCalled();
+        });
+
+        it('succesfully deinitializes the strategy', async () => {
```

Comment:
it('removes the child elements from attached to the containerId')

Reply:
Ok, thanks!